### PR TITLE
br: backup manifests related to tidb cluster in EBS backup

### LINKED ISSUE #5116: backup: take cluster manifest backup along with data.
## Feature Request

**Is your feature request related to a problem? Please describe:**

We need to save manifest information of backup cluster with backup metadata, so that, restored cluster can share the same meta information.

1. this config snapshot should be able to allow user to create a restore cluster using exactly the same setup as backup source cluster. e.g. replica #, cpu, storage of each components
2. this config snapshot should exclude backup source cluster specific info like `creationTimestamp`, `generation`, `uid`, `status` etc.
3. this snapshot should be stored in persistent storage like S3 together with data snapshot so users can easily download and use
4. Users don't expect a big performance impact when enabling config snapshot.

**Describe the feature you'd like:**

[Take cluster manifest backup along with data.](https://pingcap.zendesk.com/agent/tickets/1887)

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by WangLe1321**
### What problem does this PR solve?

backup manifests related to tidb cluster in EBS backup

Closes #5116 

### What is changed and how does it work?

retrieve the objects of `TidbMonitor`, `TidbDashboard`, `TidbClusterAutoScaler`, `TidbInitializer`, `TidbNgMonitoring`, filter them by check if their `Cluster` is equal to `tc`. Then upload them with `tc` to the remote storage. The storage path of the manifests `{backup-path}/manifests/{namespace}/{kind}/{name}.yaml`

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [x] Unit test 
- [ ] E2E test 
- [x] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [x] Need to cherry-pick to the release branch 
- [x] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Support backup manifests related to tidb cluster in EBS backup
```

### CODE REVIEW

File: pkg/backup/backup/backup_manager.go [hunk 1]
```diff
@@ -785,17 +799,92 @@ func (bm *backupManager) saveClusterMetaToExternalStorage(b *v1alpha1.Backup, cs
...
+
+	for _, manifest := range manifests {
+		if err := bm.saveManifest(b, manifest.DeepCopyObject(), externalStorage); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (bm *backupManager) saveManifest(b *v1alpha1.Backup, manifest runtime.Object, externalStorage *backuputil.StorageBackend) error {
```
Comment (BornChanger) [inline pkg/backup/backup/backup_manager.go:847]: Will cleanup clean the backed up manifest?

**Review by csuzhangxc — APPROVED** [review]

File: pkg/backup/backup/backup_manager.go (hunk 1 above)
Comment (WangLe1321) [reply pkg/backup/backup/backup_manager.go:847]: yes, they can be cleaned.

File: pkg/backup/backup/backup_manager.go [hunk 2]
```diff
@@ -785,17 +799,92 @@ func (bm *backupManager) saveClusterMetaToExternalStorage(b *v1alpha1.Backup, cs
...
+	existed, err := externalStorage.Exists(ctx, filePath)
+	if err != nil {
+		return err
 	}
-	return "", nil
+	if existed {
+		return nil
+	}
+
+	buf := bytes.NewBuffer(make([]byte, 0, 1024))
```
[the remark this answers is not in the document]
Comment (WangLe1321) [reply pkg/backup/backup/backup_manager.go:878]: just choose a larger initial capacity to reduce the number of slice expansion

**Pull request merged**